metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/TaxonomySteps.java

```diff
@@ -59,4 +55,38 @@ public void performPostOnTaxonomies(long id, String name, String ancestorKeys) t
                 .content(jsonContent.getBytes())));
     }
 
+    @When("^I request POST taxonomyTree with (.*) for species , (.*) for GENUS , (.*) for ORDER and (.*) for CLASS$")
+    public void performPostOnTaxonomyTree(String speciesUrl, String genusUrl, String orderUrl, String classUrl)
+            throws Exception {
+
+        String jsonContent = "{ " +
+                "\"taxonomySpecies\": \"" + CommonStates.getUrl(speciesUrl) + "\"," +
+                "\"taxonomyGenus\": \"" + CommonStates.getUrl(genusUrl) + "\"," +
+                "\"taxonomyOrder\": \"" + CommonStates.getUrl(orderUrl) + "\"," +
+                "\"taxonomyClass\": \"" + CommonStates.getUrl(classUrl) + "\"" +
+                "}";
+        CommonStates.setResultActions(mockMvc.perform(post("/taxonomyTrees")
+                .with(CommonStates.getRequestPostProcessor())
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonContent.getBytes())));
+    }
+
+    @When("^I request POST taxonomyTree with (.*) for species , (.*) for GENUS , (.*) for ORDER , (.*) for CLASS and " +
+            "(.*) for SUBSPECIES$")
+    public void performPostOnTaxonomyTreeWithSubspecies(String speciesUrl, String genusUrl, String orderUrl, String
+            classUrl, String subspeciesUrls)
+            throws Exception {
+
+        String jsonContent = "{ " +
+                "\"taxonomySpecies\": \"" + CommonStates.getUrl(speciesUrl) + "\"," +
+                "\"taxonomyGenus\": \"" + CommonStates.getUrl(genusUrl) + "\"," +
+                "\"taxonomyOrder\": \"" + CommonStates.getUrl(orderUrl) + "\"," +
+                "\"taxonomyClass\": \"" + CommonStates.getUrl(classUrl) + "\"," +
+                "\"taxonomySubSpecieses\":" + objectMapper.writeValueAsString(CommonStates.getUrls(subspeciesUrls))
+                + "}";
+        CommonStates.setResultActions(mockMvc.perform(post("/taxonomyTrees")
+                .with(CommonStates.getRequestPostProcessor())
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonContent.getBytes())));
+    }
```

Comment:
Do I understand correctly that if someone wants to create a new taxonomy, they will have to first create all intermediate ranks (species/genus/order/class), and only _then_ create a taxonomy? This is a problem because it makes it possible to accidentally specify an incorrect taxonomy tree. A better approach would be to request an original taxonomy ID (species or a lower level) and then create/substitute all other levels automatically

Reply:
this is creating a new taxonomy tree not creating a taxonomy. Taxonomy can be created independently but for taxonomyTree, u need the dependent ranking taxonomies

Reply:
OK, so I remember that we decided not to implement automatic construction of species/genus/order/class in _this_ ticket, but we should at least ensure that a ticket is created in the backlog for this

Reply:
yes the load taxonomy tree ticket can be used for auto load of taxonomytree